CMakeLists.txt

```diff
@@ -66,6 +65,16 @@ TARGET_COMPILE_DEFINITIONS(${PROJECT_NAME} PUBLIC
 ADD_EXECUTABLE(compiler src/main.cpp)
 TARGET_LINK_LIBRARIES(compiler ${PROJECT_NAME} ${CMAKE_DL_LIBS})
 
+ADD_CUSTOM_TARGET(
+    format
+    COMMAND clang-format -i ${COMPILER_SRCS} ${COMPILER_HEADERS}
+    )
+
 IF(TF_EXAMPLES)
     ADD_SUBDIRECTORY(${PROJECT_SOURCE_DIR}/examples)
 ENDIF()
+
+UNSET(TF_EXAMPLES CACHE)
```

Comment:
this is pretty unconventional. if i set a setting in cmake, i expect it to stick until i change it again or clear the cache myself

Reply:
Ah yes the examples should probably stay set. Given the constant need to change the fetched TF code base (tag, commit etc) it made sense to me that each call I could provide a different TF code base that I wanted.